### ISSUE TITLE
`ERR_PACKAGE_PATH_NOT_EXPORTED` error in test/config.js in Node.js nightly when requiring a subpath of @elastic/elasticsearc

### ISSUE THREAD

**Issue by trentm** [elastic/apm-agent-nodejs#2350]
Recently a test run against the latest node v17 nightly build started failing as follows:

```
$ node test/config.js
...
# disableInstrumentations
node:internal/modules/cjs/loader:488
      throw e;
      ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './package.json' is not defined by "exports" in /Users/trentm/el/apm-agent-nodejs/node_modules/@elastic/elasticsearch/package.json
    at new NodeError (node:internal/errors:371:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:416:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:669:3)
    at resolveExports (node:internal/modules/cjs/loader:482:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Test.<anonymous> (/Users/trentm/el/apm-agent-nodejs/test/config.test.js:863:19) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}

Node.js v17.0.0-nightly20210923c7da13c7c4
```

It started failing on this nightly build:

```
PASS v17.0.0-nightly202109220991dfcece/                 22-Sep-2021 07:00
FAIL v17.0.0-nightly20210923c7da13c7c4/                 23-Sep-2021 07:30
```

which means one of these commits: https://github.com/nodejs/node/compare/0991dfcece...c7da13c7c4
and almost certainly this PR https://github.com/nodejs/node/pull/40121/
which changed a DeprecationWarning, e.g.:

```
% node
Welcome to Node.js v16.9.1.
Type ".help" for more information.
> require('@elastic/elasticsearch/package.json').version
'7.14.0'
> (node:95433) [DEP0148] DeprecationWarning: Use of deprecated folder mapping "./" in the "exports" field module resolution of the package at /Users/trentm/el/apm-agent-nodejs/node_modules/@elastic/elasticsearch/package.json.
Update this package.json to use a subpath pattern like "./*".
(Use `node --trace-deprecation ...` to show where the warning was created)
```

to a runtime error:

```
% ~/tmp/node-v17.0.0-nightly20210923c7da13c7c4-darwin-x64/bin/node
Welcome to Node.js v17.0.0-nightly20210923c7da13c7c4.
Type ".help" for more information.
> require('@elastic/elasticsearch/package.json').version
Uncaught:
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './package.json' is not defined by "exports" in /Users/trentm/el/apm-agent-nodejs/node_modules/@elastic/elasticsearch/package.json
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at new NodeError (node:internal/errors:371:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:416:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:669:3)
    at resolveExports (node:internal/modules/cjs/loader:482:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at require (node:internal/modules/cjs/helpers:102:18) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

The code in question doing this, in this case, is here: https://github.com/elastic/apm-agent-nodejs/blob/60ed41332737e012a1687af5b812e86133d2ca23/test/config.test.js#L863

Note that the agent does this kind of thing in other places, including in the runtime, not just the test suite:

```
lib/agent.js
230:    // Attempt to load package.json from process.argv.
234:      pkg = require(path.join(basedir, 'package.json'))

lib/instrumentation/index.js
190:      pkg = path.join(basedir, 'package.json')
```

I'm not yet sure if this can/will affect instrumentation of `@elastic/elasticsearch`.

Note as well that the ES client already has https://github.com/elastic/elasticsearch-js/issues/1465 regarding the DeprecationWarning.

**Comment by trentm** [comment]
Looking at runtime usage of `require('.../package.json')`:

1. In "lib/instrumentation/index.js", the package.json is being read and parsed via `version = JSON.parse(fs.readFileSync(pkg)).version` so is not susceptible to this "exports" issue.
2. In "lib/agent.js", an attempt is made to `require("$CWD/package.json")` for debug logging on startup. This is guarded in a try/catch, so won't crash the app. It is a separate discussion if that require could be improved or dropped.
3. Reading of package.json in the 'read-pkg-up' dep uses fs.readFileSync and JSON.parse, so we are also good there.

tl;dr: So far this is only impacting *tests*, which is good.
